Made side-nav behaviour more Android-like

vjancik opened the pull request:
I got inspired by your talk: https://www.youtube.com/watch?v=thNyy5eYfbc
And once I started playing with the side-bar, I just HAD to make it behave exactly like on Android.

I tried to make minimal number of changes, but it's probably still a bit too complicated for your use case.
However, if that is the case, and if this example meets the performance requirements of RAIL. I think it's worthy of being a separate advanced example. That is, if you like what it does! 😄 

Please try it out!

vjancik committed:
Made side-nav behaviour more Android-like

oliverdunk commented:
Looks really cool, great work: one thing I did notice is that you aren't waiting for the touchmove event to bring out the sidebar, meaning that a single tap within 30px of the left of the screen causes it to expand. Maybe fix this so that it requires you to drag at least a pixel as in Android?

vjancik committed:
Added minimal drag distance, improved pull-back animation

vjancik commented:
Thank you!
And nice catch! Thought I had handled that.
Now that you pointed it out, I tried it on my Nexus 5 (the real thing), and it seems the default behavior is that it pulls back unless you pull out at least half of the sidebar. So I ended up doing that.
Also, the else clause animates the pullback with almost no extra code!

paullewis commented:
Dang it, I'm really torn. On the one hand, I really like these additions, but I fear that the functionality adds on a certain amount of complexity that someone new to this would struggle to parse. In many ways these examples are sort-of "here's the initial version that you can take forward", which is _exactly_ what you've done (yay!), I'm just not sure what it does to the readability for someone new.

vjancik commented:
I thought that would be the case when I finished writing it. That's why I proposed what I proposed in the first comment. I think some people will come here and ask "How can I take this further?", and this gives them that, and I think it would be a waste to throw it out.

So separate folder maybe? Mark it as "Full Implementation" or something like that.

It's up to you!

paullewis commented:
Yeah, how about this:

```
/sidenav/
  /basic/
  /android-like/
```

Then we can put a comment in a README at the root of sidenav, sidenav/basic and sidenav/android-like that explains how to understand things?

paullewis commented:
Also, totally forgot to say THANKS FOR THE PR! 🎉

oldergod commented:
@vjancik I have now a blank y scrollbar shown on desktop view, that I did not see before.

vjancik committed:
Moved to separate directory, added README

vjancik commented:
@oldergod What browser and OS are you using? I can't seem to reproduce your issue, if I understand you correctly.

@paullewis I moved it as you suggested and added a readme. If you'd like to make some changes to the readme, it will probably be quickest to do it yourself post-merging. 😄

oldergod commented:
I am talking about the scrollbar on the right here

![image](https://cloud.githubusercontent.com/assets/1767669/15851344/58bbc7ec-2cd7-11e6-9a6c-0fe07810bbf1.png)

Tested in latest FF and Chrome on Ubuntu, same result.

shotlom commented:
D

On 6:43pm, Tue, 07/06/2016 Benoît Quenaudon <notifications@github.com>
wrote:

> I am talking about the scrollbar on the right here
>
> [image: image]
> <https://cloud.githubusercontent.com/assets/1767669/15851344/58bbc7ec-2cd7-11e6-9a6c-0fe07810bbf1.png>
>
> Tested in latest FF and Chrome on Ubuntu, same result.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleChrome/ui-element-samples/pull/28#issuecomment-224217367>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYUQulKHfmWDRMS5ASPKcgEpBHkioNrks5qJS9FgaJpZM4IuZ97>
> .
>

vjancik committed:
Fixed always visible scroll bar on Linux

vjancik commented:
Sorry for the huge delay. I was busy with work and other things.

The issue is fixed and this pull request is ready to be merged. @paullewis

vjancik commented:
@paullewis Hey Paul, are you still going to merge this pull request?

surma commented:
@vjancik Could you rebase against master? We have merge conflicts.
